File: cinema-ticket-C#/Dto/Screening/CreateScreeningRequest.cs
using System.ComponentModel.DataAnnotations;

namespace CinemaTicket.Dto.Screening;

public class CreateScreeningRequest
{
    [Required]
    public long CinemaId { get; set; }

    [Required]
    public long MovieId { get; set; }

    [Required]
    public DateTime StartDateTime { get; set; }

    [Range(typeof(decimal), "0.01", "9999999999.99")]
    public decimal TicketPrice { get; set; }
}

File: Backend-Java/src/main/java/com/cinema/ticketsystem/dto/CreateScreeningRequest.java
package com.cinema.ticketsystem.dto;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CreateScreeningRequest {
    
    @NotNull(message = "Cinema ID is required")
    private Long cinemaId;
    
    @NotNull(message = "Movie ID is required")
    private Long movieId;
    
    @NotNull(message = "Start date time is required")
    private LocalDateTime startDateTime;
    
    @NotNull(message = "Ticket price is required")
    private java.math.BigDecimal ticketPrice;
}
